File: services/ordering/src/main/java/io/wkrzywiec/fooddelivery/ordering/infra/store/OrderingEventClassTypeProvider.java
package io.wkrzywiec.fooddelivery.ordering.infra.store;

import io.wkrzywiec.fooddelivery.commons.infra.store.DomainEvent;
import io.wkrzywiec.fooddelivery.commons.infra.store.EventClassTypeProvider;
import io.wkrzywiec.fooddelivery.ordering.domain.OrderingEvent;
import lombok.extern.slf4j.Slf4j;

@Slf4j
class OrderingEventClassTypeProvider implements EventClassTypeProvider {

    @Override
    public Class<? extends DomainEvent> getClassType(String type) {
        return switch (type) {
            case "OrderCreated" -> OrderingEvent.OrderCreated.class;
            case "OrderCanceled" -> OrderingEvent.OrderCanceled.class;
            case "OrderInProgress" -> OrderingEvent.OrderInProgress.class;
            case "TipAddedToOrder" -> OrderingEvent.TipAddedToOrder.class;
            case "OrderCompleted" -> OrderingEvent.OrderCompleted.class;
            default -> {
                log.error("There is not logic for mapping {} event from a store", type);
                yield null;
            }
        };
    }
}

File: services/ordering/src/main/kotlin/io/wkrzywiec/fooddelivery/ordering/infra/store/OrderingEventClassTypeProvider.kt
package io.wkrzywiec.fooddelivery.ordering.infra.store

import io.github.oshai.kotlinlogging.KotlinLogging
import io.wkrzywiec.fooddelivery.commons.infra.store.DomainEvent
import io.wkrzywiec.fooddelivery.commons.infra.store.EventClassTypeProvider
import io.wkrzywiec.fooddelivery.ordering.domain.OrderingEvent
import lombok.extern.slf4j.Slf4j

private val logger = KotlinLogging.logger {}

@Slf4j
internal class OrderingEventClassTypeProvider : EventClassTypeProvider {
    override fun getClassType(type: String): Class<out DomainEvent>? {
        return when (type) {
            "OrderCreated" -> OrderingEvent.OrderCreated::class.java
            "OrderCanceled" -> OrderingEvent.OrderCanceled::class.java
            "OrderInProgress" -> OrderingEvent.OrderInProgress::class.java
            "TipAddedToOrder" -> OrderingEvent.TipAddedToOrder::class.java
            "OrderCompleted" -> OrderingEvent.OrderCompleted::class.java

            else -> {
                logger.error { "${"There is not logic for mapping {} event from a store"} $type" }
                return null
            }
        }
    }
}
